BLE Indoor Positioning/src/test/java/com/nexenio/bleindoorpositioning/location/provider/LocationPredictorTest.java

```diff
@@ -0,0 +1,102 @@
+package com.nexenio.bleindoorpositioning.location.provider;
+
+import com.nexenio.bleindoorpositioning.location.Location;
+import com.nexenio.bleindoorpositioning.location.LocationPredictor;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Created by leon on 14.02.18.
+ */
+public class LocationPredictorTest {
+
+    public static final double HUMAN_WALKING_SPEED = 1.388889; // meters per second
+
+    public static final Location FIRST_POINT_OF_LINE = new Location(52.513588, 13.392995);
+    public static final Location SECOND_POINT_OF_LINE = new Location(52.513632, 13.392986);
+    public static final Location THIRD_POINT_OF_LINE = new Location(52.513675, 13.392979);
+    public static final Location FOURTH_POINT_OF_LINE = new Location(52.513720, 13.392971);
+    public static final Location FIFTH_POINT_OF_LINE = new Location(52.513764, 13.392962);
+
+    @Test
+    public void predictLocationFromLocations_locationsInLine_accuratePrediction() throws Exception {
+        List<Location> lineOnGendarmenmarkt = new ArrayList<>(Arrays.asList(
+                new Location(FIRST_POINT_OF_LINE), new Location(SECOND_POINT_OF_LINE),
+                new Location(THIRD_POINT_OF_LINE), new Location(FOURTH_POINT_OF_LINE)
+        ));
+        lineOnGendarmenmarkt = setupTimestampsForLocations(lineOnGendarmenmarkt);
+        Location predictedLocation = LocationPredictor.predict(lineOnGendarmenmarkt, TimeUnit.SECONDS.toMillis(1));
+        double delta = predictedLocation.getDistanceTo(FIFTH_POINT_OF_LINE);
+        assertEquals(0, delta, 5);
+    }
+
+    @Test
+    public void predictLocationFromLocations_locationsInCircle_accuratePrediction() throws Exception {
+        long timestampDelta = 500;
+        double distanceToNextLocation = HUMAN_WALKING_SPEED / TimeUnit.MILLISECONDS.toSeconds(timestampDelta);
+        List<Location> walkInACircle = new ArrayList<>();
+        Location nextLocation;
+        Location lastLocation = null;
+        for (int angle = 0; angle < 360; angle += 10) {
```

Comment:
Move this to a new method which takes a list of points as input and calculates everything automatically for arbitrary location sequences.